src/hooks/useInitialLoadingStatus.ts

```diff
@@ -0,0 +1,57 @@
+import { atom, useAtom } from 'jotai';
+
+type InitialLoadingStatus =
+    | 'READY' // initial sync for the drive has been finished (either failed or succeeded)
+    | 'LOADING' // remote drive is being loaded but not synced yet
+    | 'SYNCING'; // remote drive is syncing
+
+const DEFAULT_DRIVES_URL = import.meta.env.VITE_DEFAULT_DRIVES_URL || undefined;
+const defaultDrives = DEFAULT_DRIVES_URL ? DEFAULT_DRIVES_URL.split(',') : [];
+
+const defaultInitialState = defaultDrives.map(driveUrl => ({
+    url: driveUrl,
+    initialSyncStatus: 'LOADING' as InitialLoadingStatus,
+}));
+
+const drivesInitialSyncStatusAtom = atom(defaultInitialState);
+
+export const useInitialLoadingStatus = () => {
+    const [initialLoadingStatus, setInitialLoadingStatus] = useAtom(
+        drivesInitialSyncStatusAtom,
+    );
+
+    const setDriveInitialLoadingStatus = (
+        url: string,
+        status: InitialLoadingStatus,
+    ) =>
+        setInitialLoadingStatus(prevState =>
+            prevState.map(drive =>
+                drive.url === url
+                    ? { ...drive, initialSyncStatus: status }
+                    : drive,
+            ),
+        );
+
+    const getDriveInitialLoadingStatus = (driveId?: string) => {
```

Comment:
Returns the initial sync status for a given drive. If no status is found, it defaults to `READY`. This way, if the user attempts to visit a URL with a drive that is not part of the default drives, no loading screen is displayed.